snuba_sdk/conditions.py

```diff
@@ -133,17 +133,16 @@ def validate(self) -> None:
 @dataclass(frozen=True)
 class And(BooleanCondition):
     op: BooleanOp = field(init=False, default=BooleanOp.AND)
-    conditions: Sequence[Union[BooleanCondition, Condition]] = field(
-        default_factory=list
-    )
+    conditions: ConditionSet = field(default_factory=list)
 
 
 @dataclass(frozen=True)
 class Or(BooleanCondition):
     op: BooleanOp = field(init=False, default=BooleanOp.OR)
-    conditions: Sequence[Union[BooleanCondition, Condition]] = field(
-        default_factory=list
-    )
+    conditions: ConditionSet = field(default_factory=list)
+
+
+ConditionSet = Sequence[Union[BooleanCondition, Condition]]
```

Comment:
ConditionSet implies uniqueness but this is not a `set`, it's a sequence
```suggestion
ConditionSequence = Sequence[Union[BooleanCondition, Condition]]
```

Reply:
As a side note, I do find it strange that `BooleanCondition`s and `Condition`s are different things but that's out of scope for this PR